### PR TITLE
adding 2020-10-23-kubernetes-webhooks-made-easy-with-openfaas post

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

## Description
This PR includes a new blog post about how can we use Kubernetes Validating Admission Webhooks and OpenFaaS together.

## Types of changes

- [x] New blog post
- [ ] Updating an existing blog post
- [ ] Updating part of the page page
- [ ] Adding a new web-page

## Checklist:

- [x] I have given attribution for any images I have used and have permission to use them under Copyright law
- [x] My code follows the [writing-style of the publication](README.md) and I have checked this
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`

### CODE REVIEW

**Comment by alexellis** [comment]
This is very cool.

I noticed that the code example has the same function hard-coded for every event. Can you explain how users can change that, and maybe show two separate functions that do different things, both of which get evaluated for each event?

**Comment by alexellis** [comment]
Please have a read of ["Editorial / style guidelines"](https://github.com/openfaas/openfaas.github.io#editorial--style-guidelines) before we review, and check you've hit all the points like image size etc.

Looking forward to trying this myself. 

Also - how would people use this on ARM? If they can't yet, you should say that somewhere, as people will try.

**Comment by developer-guy** [comment]
> This is very cool.
> 
> I noticed that the code example has the same function hard-coded for every event. Can you explain how users can change that, and maybe show two separate functions that do different things, both of which get evaluated for each event?

We updated the blog post according to this review, we added a new section for the cloning project and also about how can they extend the project. see: e1ac2f5

File: _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+Webhook Admission Server is just plain http server that adhere to Kubernetes API. For each Pod create request to the apiserver(I said Pod because we specify which kind of resources that we consider while registering our webhook to the apiserver using [ValidatingWebhookConfiguration](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#validatingwebhookconfiguration-v1-admissionregistration-k8s-io) resource) the ValidatingAdmissionWebhook sends an admissionReview([API](https://github.com/kubernetes/kubernetes/blob/release-1.19/pkg/apis/admission/types.go) for reference) to the relevant webhook admission server. The webhook admission server gathers information like object, oldobject, and userInfo from admissionReview's AdmissionRequest and sends AdmissionRequest to the serverless function through the OpenFaaS Gateway. The function checks the required labels exist on Pod and determines the request either valid or not and then sends back the AdmissionResponse whose Allowed and Result fields are filled with the admission decision to the webhook admission server then the webhook admission servers sends back a admissionReview to the apiserver.
```
Comment (alexellis) [inline _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md:41]: adheres

File: _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+<p align="center">
+<img height="128" src="/images/2020-10-23-k8s-validatingwebhook-openfaas/heart.png">
+</p>
+
+<p align="center">
+<img height="128" src="/images/openfaas/kubernetes.png">
+</p>
+
+## Introduction to Kubernetes Admission webhooks
```
Comment (alexellis) [inline _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md:33]: In this section please give a short 1-liner on why you used openfaas for this over doing it with the microservice on its own.

File: _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md [hunk 3]
```diff
@@ -0,0 +1,265 @@
+---
+title: "Kubernetes Webhooks made easy with OpenFaaS"
+description: "Extending the Kubernetes with Admission webhook using OpenFaaS Serverless Functions"
+date: 2020-10-23
+image: /images/2020-10-github-actions/lego.jpg
```
Comment (alexellis) [inline _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md:5]: Please pick your own background for the post and resize as per the README - it gives dimensions

Unplash and pexels are both good websites for royalty free stock photography.

File: _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+categories:
+  - arkade
+  - kubectl
+  - faas-cli
+  - kind
+  - k8s
```
Comment (alexellis) [inline _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md:11]: Let's drop kind + k8s.. too many tags.

File: _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md [hunk 5]
```diff
@@ -0,0 +1,265 @@
+---
+title: "Kubernetes Webhooks made easy with OpenFaaS"
+description: "Extending the Kubernetes with Admission webhook using OpenFaaS Serverless Functions"
```
Comment (alexellis) [inline _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md:3]: `In this post you'll learn how to write Kubernetes Admission webhooks using OpenFaaS functions`

File: _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+date: 2020-10-23
```
Comment (alexellis) [inline _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md:4]: Put the date to 27th October please and rename the file with `git mv` so the newer date.

File: _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+
+* Kubernetes API -> Webhook w/TLS -> OpenFaaS Gateway (w HTTP)
```
Comment (alexellis) [inline _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md:43]: One more `->` to the Function

File: _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md (hunk 1 above)
Comment (developer-guy) [reply _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md:41]: what should I do to them?

**Review by alexellis — CHANGES_REQUESTED** [review]
Please see my comments.

File: _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md (hunk 7 above)
Comment (developer-guy) [reply _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md]: done.

File: _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md (hunk 6 above)
Comment (developer-guy) [reply _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md]: done.

File: _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md (hunk 5 above)
Comment (developer-guy) [reply _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md]: done.

File: _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md (hunk 4 above)
Comment (developer-guy) [reply _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md]: done.

File: _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md (hunk 3 above)
Comment (developer-guy) [reply _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md]: done.

File: _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md (hunk 2 above)
Comment (developer-guy) [reply _posts/2020-10-23-kubernetes-webhooks-made-easy-with-openfaas.md]: done.

**Pull request closed without merging**